### PR TITLE
Added UI AK list associated hosts test

### LINKED ISSUE #3919: Test to Automate: Activation Key should list only those content host which were registered via the key
`Steps to Reproduce:`
1. register some Content Hosts
2. Create any activation key
3. Click to the AK -> Associations -> Content Hosts

`Actual results:`
List of all Content Hosts is shown.

`Expected results:`
Just Content Hosts activated via the AK shall be listed.

More info: https://bugzilla.redhat.com/show_bug.cgi?id=1372826

### PR DESCRIPTION
**Pull request by abalakh**
Closes #3919

Test results:
```python
py.test tests/foreman/ui/test_activationkey.py -v -k 'test_positive_list_hosts'
================================================ test session starts ================================================
platform linux2 -- Python 2.7.10, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/andrii/env/bin/python
cachedir: .cache
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
collected 33 items 

tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_list_hosts PASSED

================================================ 32 tests deselected ================================================
===================================== 1 passed, 32 deselected in 416.58 seconds =====================================
```

### CODE REVIEW

File: robottelo/ui/activationkey.py [hunk 1]
```diff
@@ -59,6 +59,16 @@ def search_key_subscriptions(self, ak_name, subscription_name):
             (strategy, value % subscription_name))
         return element
 
+    def search_host(self, ak_name, host_name):
+        """Search for host activated by selected activation key."""
+        self.click(self.search(ak_name))
```
Comment (svtkachenko) [inline robottelo/ui/activationkey.py]: Why not to use self.search_and_click? :)

Comment (abalakh) [inline robottelo/ui/activationkey.py]: Sure. TBH wasn't aware of it.

Comment (abalakh) [inline robottelo/ui/activationkey.py]: We should probably improve our PR naming, as the PR with corresponding change was called "Added *** tests" so as i wasn't reviewer i never checked the code and had no idea new base helper was introduced.

**Comment by oshtaier** [comment]
API+CLI+UI in one module? Could we implement that in any other way. Like actually doing steps through UI? Optimization is really important thing, but we should not forget about some isolation

**Comment by abalakh** [comment]
@oshtaier registering a host will require some cli commands anyways, also helpers used here (like `setup_org_for_a_rh_repo`) are implemented for cli only. However, i can easily drop API part

**Comment by sghai** [comment]
ACK.. test looks good to me. The validation or assertion should be done via UI only and that's how the test is automated. Thank you @abalakh for quickly picking up this test.

**Comment by abalakh** [comment]
@oshtaier PR updated, dropped api calls from the test. Not sure in what way it's better as the AK tests module still uses nailgun a lot.
Test results:
```python
py.test tests/foreman/ui/test_activationkey.py -v -k 'test_positive_list_hosts'
================================================ test session starts ================================================
platform linux2 -- Python 2.7.10, pytest-3.0.3, py-1.4.31, pluggy-0.4.0 -- /home/andrii/env/bin/python
cachedir: .cache
rootdir: /home/andrii/workspace/robottelo, inifile: 
plugins: xdist-1.13.1, repeat-0.2
collected 33 items 

tests/foreman/ui/test_activationkey.py::ActivationKeyTestCase::test_positive_list_hosts PASSED

================================================ 32 tests deselected ================================================
===================================== 1 passed, 32 deselected in 458.79 seconds =====================================
```

**Comment by renzon** [comment]
@abalakh I have added this question regarding using API on UI tests: #3928

**Comment by renzon** [comment]
ACK

**Pull request merged**